core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroup.java

```diff
@@ -91,6 +91,10 @@ protected HealthCheckedEndpointGroup(ClientFactory clientFactory,
     protected void init() {
         checkState(scheduledCheck == null, "init() must only be called once.");
 
+        if (delegate instanceof DynamicEndpointGroup) {
+            ((DynamicEndpointGroup) delegate).initialEndpointsFuture().join();
```

Comment:
Should we pull this method up to `EndpointGroup`?

Reply:
Yeah I was thinking of it - it would require quite a bit of refactoring so thinking of doing it in another PR. What do you think?

Reply:
Went ahead and moved it in this one.